cli/src/main/java/com/devonfw/tools/ide/tool/sonar/Sonar.java

```diff
@@ -0,0 +1,100 @@
+package com.devonfw.tools.ide.tool.sonar;
+
+import java.nio.file.Path;
+import java.util.Properties;
+import java.util.Set;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.property.EnumProperty;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.tool.java.Java;
+import com.devonfw.tools.ide.tool.mvn.Mvn;
+import com.devonfw.tools.ide.util.PropertiesFileUtil;
+
+public class Sonar extends LocalToolCommandlet {
+
+  public final EnumProperty<SonarCommand> command;
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}. method.
+   */
+  public Sonar(IdeContext context) {
+
+    super(context, "sonar", Set.of(Tag.CODE_QA));
+
+    this.command = add(new EnumProperty<>("", true, "command", SonarCommand.class));
+    add(this.arguments);
+  }
+
+  @Override
+  protected void initProperties() {
+
+    // Empty on purpose
+  }
+
+  @Override
+  public boolean install(boolean silent) {
+
+    getCommandlet(Java.class).install();
+    return super.install(silent);
```

Comment:
you may need to override postinstall and set the executable flags for the linux/macOS scripts since the download is a ZIP file and ZIP cannot represent executable flags.
https://github.com/devonfw/ide-urls/blob/master/sonar/sonar/10.4.0.87286/urls

I just checked the download from the sonarqube website and it also only gives me a ZIP. Broken by design of sonarqube. So we need to set the X-flag manually on Linux / Mac.